inlang/source-code/paraglide/paraglide-js/src/compilerV2/compile.ts

```diff
@@ -0,0 +1,143 @@
+import { compileBundle } from "./compileBundle.js"
+import { telemetry } from "../services/telemetry/implementation.js"
+import { i } from "../services/codegen/identifier.js"
+import { getStackInfo } from "../services/telemetry/stack-detection.js"
+import { getPackageJson } from "../services/environment/package.js"
+import { createRuntime } from "./runtime.js"
+import { createRegistry } from "./registry.js"
+import { lookup } from "@inlang/sdk"
+import { type BundleNested, type ProjectSettings } from "@inlang/sdk2"
+import fs from "node:fs/promises"
+import * as prettier from "prettier"
+import { escapeForSingleQuoteString } from "~/services/codegen/escape.js"
+
+const ignoreDirectory = `# ignore everything because the directory is auto-generated by inlang paraglide-js
+# for more info visit https://inlang.com/m/gerre34r/paraglide-js
+*
+`
+
+export type CompileOptions = {
+	bundles: Readonly<BundleNested[]>
+	settings: Pick<ProjectSettings, "baseLocale" | "locales">
+	projectId: string | undefined
+}
+
+const defaultCompileOptions = {
+	projectId: undefined,
+} satisfies Partial<CompileOptions>
+
+/**
+ * A compile function takes a list of messages and project settings and returns
+ * a map of file names to file contents.
+ *
+ * @example
+ *   const output = compile({ messages, settings })
+ *   console.log(output)
+ *   >> { "messages.js": "...", "runtime.js": "..." }
+ */
+export const compile = async (args: CompileOptions): Promise<Record<string, string>> => {
+	const opts = {
+		...defaultCompileOptions,
+		...args,
+	}
+
+	//Maps each language to it's fallback
+	//If there is no fallback, it will be undefined
+	const fallbackMap = getFallbackMap(opts.settings.locales, opts.settings.baseLocale)
+
+	const resources = opts.bundles.map((bundle) => compileBundle(bundle, fallbackMap))
+
+	const indexFile = [
+		"/* eslint-disable */",
+		'import { languageTag } from "./runtime.js"',
+		opts.settings.locales
+			.map((locale) => `import * as ${i(locale)} from "./messages/${locale}.js"`)
```

Comment:
I suggest renaming `i` to `jsIdentifier` or similar. 

I was wondering where `i` was coming from and what `i` is. An index? Why index over the locale? Using `jsIdentifier` would improve the comprehension.